dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/DefaultTrackerValidationServiceTest.java

```diff
@@ -48,116 +50,107 @@
 import org.hisp.dhis.user.User;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.extension.ExtendWith;
-import org.mockito.InjectMocks;
-import org.mockito.Mock;
-import org.mockito.junit.jupiter.MockitoExtension;
-import org.springframework.test.util.ReflectionTestUtils;
 
-@ExtendWith( MockitoExtension.class )
 class DefaultTrackerValidationServiceTest
 {
 
-    @InjectMocks
-    private DefaultTrackerValidationService trackerValidationService;
+    private DefaultTrackerValidationService service;
```

Comment:
Why are we moving away from @mock annotation?

Reply:
@jbee mentioned that this approach has the drawback of creating mocks for every test even if they are not used. Without the `@Mock` we can target them more specifically per test. In this test class we are only saving one instantiation (`hook2` which is only needed in one test).

We could talk about the pros/cons in a backend team meeting. Not sure if the platform team has any actual plan for moving away from it.

Reply:
From my experience the `@Mock` fields have a tendency to cement the status quo of the test setup because it becomes very hard to tell what is done for which test case/scenario and if it is now after some changes unnecessary and could be removed. This becomes more clear if you use `mock(Class)` as you can scope this accordingly and have some of those setup in a general `setUp()` method and others that belong to a specific scenario just done in that scenario.

Reply:
Nice :)
I was just asking to understand the reasons behind that.